### PR TITLE
Reason Severity Translation Workbench changes

### PR DESCRIPTION
**Pull request by visoft**
This PR is the new Reason Severity workflow. It adds on to the Reason modal (only shown to reviewers who change a translation). Now, reviewers must choose a severity along with one or more reasons. It contains a migration for adding a new column to the `translation_changes` table to handle the reason severity.

### CODE REVIEW

File: app/assets/javascripts/application/translation_workbench.js.coffee [hunk 1]
```diff
@@ -185,6 +188,17 @@ class TranslationItem
 
       context.reasons = reasons
 
+      [first, ..., last] = @translation.changes
```
Comment (dliusquare) [inline app/assets/javascripts/application/translation_workbench.js.coffee]: Is `last` used anywhere?

File: app/assets/javascripts/application/translation_workbench.js.coffee [hunk 2]
```diff
@@ -130,6 +131,8 @@ class TranslationItem
 
   # @private
   build: ->
+    severity_texts = ['Minor', 'Neutral', 'Major', 'Critical']
```
Comment (dliusquare) [inline app/assets/javascripts/application/translation_workbench.js.coffee]: Better to move these strings to `en.yml`.

File: app/assets/javascripts/application/translation_workbench.js.coffee (hunk 1 above)
Comment (visoft) [reply app/assets/javascripts/application/translation_workbench.js.coffee]: Good point, I’ll remove it.

File: app/assets/javascripts/application/translation_workbench.js.coffee (hunk 2 above)
Comment (visoft) [reply app/assets/javascripts/application/translation_workbench.js.coffee]: These actually come from an enum on `Translation`. They probably should be moved to `translation_changes` though. I'll make that change, and should I pull the enum values from there or still use `en.yml`?

Comment (visoft) [reply app/assets/javascripts/application/translation_workbench.js.coffee]: I moved the enum to `TranslationChange` and made adjustments for that, and used the enum in the Translation Workbench.

File: app/assets/javascripts/application/translation_workbench.js.coffee.erb [hunk 3]
```diff
@@ -131,7 +131,7 @@ class TranslationItem
 
   # @private
   build: ->
-    severity_texts = ['Minor', 'Neutral', 'Major', 'Critical']
+    severity_texts = <%= TranslationChange.reason_severities.map {|s| s[0].titlecase} %>
```
Comment (dliusquare) [inline app/assets/javascripts/application/translation_workbench.js.coffee.erb]: I mean the display string should not be `converted` from enum. (I know Shuttle does it several times in other places).
The code can use enum or strings to differentiate the severities. But the display strings need to be loaded from en.yml.

Anyway, I will approve and merge the PR for now.

**Pull request merged**